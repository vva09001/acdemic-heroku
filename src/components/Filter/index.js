import React from "react";
import PropTypes from "prop-types";

export const StudentStatus = ({ list, filter }) => {
  if (list.length > 0) {
    const listStatus = list.map(e => {
      return (
        <li key={e.ssId} onClick={() => filter(e.ssId)}>
          {e.ssName.split("")}
        </li>
      );
    });
    return (
      <React.Fragment>
        <ul className="filter studentStatus">
          <li onClick={() => filter("")}>Tất cả</li>
          {listStatus}
        </ul>
      </React.Fragment>
    );
  }
  return <React.Fragment />;
};

export const AllProgram = ({ list, filter }) => {
  if (list.length > 0) {
    const listProgram = list.map(e => {
      return (
        <li key={e.programmeId} onClick={() => filter(e.programmeId)}>
          {e.programmeName}
        </li>
      );
    });
    return (
      <React.Fragment>
        <ul className="filter allProgram">
          <li onClick={() => filter("")}>Tất cả</li>
          {listProgram}
        </ul>
      </React.Fragment>
    );
  }
  return <React.Fragment />;
};

export const ClassStatus = ({ list, filter }) => {
  if (list.length > 0) {
    const listStatus = list.map(e => {
      return (
        <li key={e.csId} onClick={() => filter && filter(e.csId)}>
          {e.csName}
        </li>
      );
    });
    return (
      <React.Fragment>
        <ul className="filter classStatus">
          <li onClick={() => filter("")}>Tất cả</li>
          {listStatus}
        </ul>
      </React.Fragment>
    );
  }
  return <React.Fragment />;
};

StudentStatus.propTypes = {
  list: PropTypes.array
};
StudentStatus.defaultProps = {
  list: []
};

AllProgram.propTypes = {
  list: PropTypes.array
};
AllProgram.defaultProps = {
  list: []
};

ClassStatus.propTypes = {
  ist: PropTypes.array
};
ClassStatus.defaultProps = {
  list: []
};
