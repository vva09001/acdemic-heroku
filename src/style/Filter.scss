$primaryColor: #203e7d;
$secondColor: #0060ae;

.filter {
  z-index: 1;
  position: absolute;
  display: none;
  background: #fff;
  box-shadow: 3px 3px 5px 0px #777;
  padding: 0px;
  li {
    list-style: none;
    font-size: 10px;
    font-weight: 400;
    padding: 5px 10px;
    cursor: pointer;
    &:hover {
      color: #fff;
      background: $primaryColor;
    }
  }
}

.status:hover .studentStatus {
  display: block
}

.program:hover .allProgram {
  display: block
}

.classesStatus:hover .classStatus {
  display: block
}