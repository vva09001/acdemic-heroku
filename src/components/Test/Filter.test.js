import React from "react";
import { create } from "react-test-renderer";
import { StudentStatus, AllProgram } from "../Filter";

describe("StudentStatus Component", () => {
  it("Test StudentStatus component it work", () => {
    const component = create(<StudentStatus />).toJSON();
    expect(component).toMatchSnapshot();
  });
  exports[`Rerender StudentStatus component`] = (
    <StudentStatus list={[Array]} filterByStudentStatus={[Function]} />
  );

  it("Test AllProgram component it work", () => {
    const component = create(<AllProgram />).toJSON();
    expect(component).toMatchSnapshot();
  });
  exports[`Rerender AllProgram component`] = (
    <AllProgram list={undefined} filterByProgram={[Function]} />
  );
});
