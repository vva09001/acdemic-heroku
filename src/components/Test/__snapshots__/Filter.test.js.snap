// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentStatus Component Test AllProgram component it work 1`] = `null`;

exports[`StudentStatus Component Test StudentStatus component it work 1`] = `null`;
